strongbox-commons/src/main/java/org/carlspring/strongbox/util/JarFileClassLoader.java

```diff
@@ -0,0 +1,41 @@
+package org.carlspring.strongbox.util;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Enumeration;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class JarFileClassLoader
+{
+
+    public static void loadClasses(ClassLoader classLoader,
+                                   String pathToJar)
+            throws IOException, ClassNotFoundException
+    {
+
+        final URLClassLoader cl = ClassLoaderUtils.getURLClassLoader(classLoader);
+        ClassLoaderUtils.addURLToURLClassLoader(cl, new URL[]{ new URL("jar:file:" + pathToJar + "!/") });
+
+        final JarFile jarFile = new JarFile(pathToJar);
```

Comment:
@fuss86 : I think you might like to have another look at this and make sure the stream gets closed properly, so that it doesn't leak resources and gradually cause an `OutOfMemoryError`.